src/Corax/Indexing/IndexedField.cs

```diff
@@ -38,10 +41,11 @@ public struct SpatialEntry
     public readonly bool HasSuggestions;
     public readonly bool ShouldStore;
     public readonly SupportedFeatures SupportedFeatures;
+    public readonly bool IsVirtual;
```

Comment:
What is the meaning of "virtual"? Does it apply to dynamic fields?

Reply:
Yes, this is strictly for dynamic field but not for every dynamic field. This is basically a shallow copy of known (which we know from compiling/analyzing index definition) field for dynamic which exactly the same name. Idea here is to have reference to in-memory structures which we build to gather all data to insert to the trees since we've two different paths to write to exactly the same trees.

More detail explanation is here: https://github.com/ravendb/ravendb/pull/18545#discussion_r1608054625